### PR TITLE
Issue #4750: Consolidate watchdog severity levels.

### LINKED ISSUE #4750: Use watchdog_severity_levels() instead of duplicating information
A function exists that pairs the various watchdog severity levels with their human-readable names: [`watchdog_severity_levels()`](https://api.backdropcms.org/api/backdrop/core%21includes%21common.inc/function/watchdog_severity_levels/1)

And yet there are places where that information in duplicated unnecessarily:
- [`hook_watchdog()`](https://api.backdropcms.org/api/backdrop/core%21modules%21system%21system.api.php/function/hook_watchdog/1)
- [`system_logging_settings()`](https://api.backdropcms.org/api/backdrop/core%21modules%21system%21system.admin.inc/function/system_logging_settings/1)
- [`dblog_overview()`](https://api.backdropcms.org/api/backdrop/core%21modules%21dblog%21dblog.admin.inc/function/dblog_overview/1)

We should consolidate...

### PR DESCRIPTION
**Pull request by ghost**
Fixes https://github.com/backdrop/backdrop-issues/issues/4750

### CODE REVIEW

File: core/includes/common.inc [hunk 1]
```diff
@@ -8495,25 +8495,81 @@ function backdrop_parse_info_format($data, $process_sections = FALSE) {
...
-    WATCHDOG_WARNING => t('warning'),
-    WATCHDOG_NOTICE  => t('notice'),
-    WATCHDOG_INFO    => t('info'),
-    WATCHDOG_DEBUG   => t('debug'),
-    WATCHDOG_DEPRECATED => t('deprecated'),
+function watchdog_severity_levels($return = 'human') {
+  $levels = array(
+    WATCHDOG_EMERGENCY => array(
+      'machine' => 'emergency',
+      'human' => t('Emergency'),
```
Comment (klonos) [inline core/includes/common.inc]: We usually call such things `name`/`label`.

Comment (ghost) [reply core/includes/common.inc]: Nice, thanks!

File: core/includes/common.inc [hunk 2]
```diff
@@ -8507,25 +8507,81 @@ function backdrop_parse_info_format($data, $process_sections = FALSE) {
...
+      'label' => t('Deprecated'),
+      'description' => t('Deprecated function/feature notices'),
+    ),
   );
+
+  switch ($return) {
+    case 'name':
+    case 'label':
+    case 'description':
+      return array_combine(array_keys($levels), array_column($levels, $return));
```
Comment (klonos) [inline core/includes/common.inc:8655]: @BWPanda this is micro-optimizing, so feel free to ignore, but various Stack Exchange articles claim that loops are less expensive than using `array_combine()`/`array_keys()`/`array_column()`. So perhaps consider switching this to a `foreach` with simple `unset()`s in it?

Comment (quicksketch) [reply core/includes/common.inc:8655]: I don't think this needs to be changed, this is a very small array.

File: core/includes/bootstrap.inc [hunk 3]
```diff
@@ -2290,17 +2290,8 @@ function watchdog_severity_enabled($severity = WATCHDOG_NOTICE) {
...
- *   - WATCHDOG_ALERT: Alert, action must be taken immediately.
- *   - WATCHDOG_CRITICAL: Critical conditions.
- *   - WATCHDOG_ERROR: Error conditions.
- *   - WATCHDOG_WARNING: Warning conditions.
- *   - WATCHDOG_NOTICE: (default) Normal but significant conditions.
- *   - WATCHDOG_INFO: Informational messages.
- *   - WATCHDOG_DEBUG: Debug-level messages.
- *   - WATCHDOG_DEPRECATED: Use of deprecated functions or features.
+ *   The severity of the message; see watchdog_severity_levels() for
+ *   possible values.
```
Comment (quicksketch) [inline core/includes/bootstrap.inc]: I think this docblock is worth duplicating the list of levels, since `watchdog()` is the primary function called where the value needs to be specified. So let's revert this change while leaving the current `@see watchdog_severity_levels()` in place.

```suggestion
 *   The severity of the message; one of the following values as defined in
 *   @link http://www.faqs.org/rfcs/rfc3164.html RFC 3164: @endlink
 *   - WATCHDOG_EMERGENCY: Emergency, system is unusable.
 *   - WATCHDOG_ALERT: Alert, action must be taken immediately.
 *   - WATCHDOG_CRITICAL: Critical conditions.
 *   - WATCHDOG_ERROR: Error conditions.
 *   - WATCHDOG_WARNING: Warning conditions.
 *   - WATCHDOG_NOTICE: (default) Normal but significant conditions.
 *   - WATCHDOG_INFO: Informational messages.
 *   - WATCHDOG_DEBUG: Debug-level messages.
 *   - WATCHDOG_DEPRECATED: Use of deprecated functions or features.
```
